### PR TITLE
WAITP-1502: code guidelines

### PR DESCRIPTION
**Pull request by acdunham**
### Changes:
- Added contributing guidelines
- Updated code of conduct to latest version

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+Thank you for taking the time to contribute, and welcome to the Tupaia!
+We value your expertise, insights, and commitment to making Tupaia a global good. This
+document is a guide on how to contribute to Tupaia effectively, ensuring it remains a
+cohesive, maintainable, and globally valuable product.
+
+## Code of conduct
+
+This project and everyone participating in it is governed by the
+[BES Open Source Code of Conduct](code-of-conduct.md).
+By participating, you are expected to uphold this code. Please report unacceptable behaviour to [opensource@bes.au](opensource@bes.au).
```
Comment (edmofro) [inline .github/CONTRIBUTING.md:12]: Have created this email address, forwards to me

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+## Copyright Attribution
+
+All open-source contributors to Tupaia will be required to sign a Contributor License Agreement (CLA)
+before BES can accept your contribution into the project code. This is done so that we can continue
+to maintain the codebase covered by a single license. Signing a CLA can be done through GitHub
+before merging in your first pull request.
```
Comment (edmofro) [inline .github/CONTRIBUTING.md:19]: It really can? That's cool.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+## Our Philosophy
+
+Our core philosophy is to maintain the long-term sustainability of Tupaia. We, at BES work hard
+to maintain a cohesive vision for the project. While we encourage collaboration and contributions,
+we also want to avoid fragmentation that could compromise Tupaia's usability, maintainability,
+and value. We believe that the best way to contribute is to align your efforts with our existing
+product roadmap.
```
Comment (edmofro) [inline .github/CONTRIBUTING.md:27]: I recognise this as my own writing! Nice. Must've come through from Slab.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+When your contribution is ready for review, create a pull request. A BES developer will review
+your changes and provide feedback. If your pull request has been approved by a reviewer, it
+will then go through a round of internal testing. Once your changes have passed testing, they
+can be merged into main and be included in an upcoming Tupaia release.
+
+### Note on Forking
+
+While Tupaia is open-source, there is always the possibility for forking the repository, which
+we strongly recommend against it. We will not provide any support for forked versions of Tupaia.
```
Comment (edmofro) [inline .github/CONTRIBUTING.md:107]:
```suggestion
we strongly recommend against. We will not provide any support for forked versions of Tupaia.
```

File: .github/code-of-conduct.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+## Enforcement Responsibilities
+
+Community leaders are responsible for clarifying and enforcing our standards of acceptable behaviour and will take appropriate and fair corrective action in response to any behaviour that they deem inappropriate, threatening, offensive, or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, and will communicate reasons for moderation decisions when appropriate.
```
Comment (edmofro) [inline .github/code-of-conduct.md:35]:
```suggestion
Community leaders have the right and responsibility to remove, edit, or reject comments, commits, code, docs site edits, issues, and other contributions that are not aligned to this Code of Conduct, and will communicate reasons for moderation decisions when appropriate.
```

File: .github/code-of-conduct.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+Community Impact: A serious violation of community standards, including sustained inappropriate behaviour.
+
+Consequence: A temporary ban from any sort of interaction or public communication with the community for a specified period of time. No public or private interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct, is allowed during this period. Violating these terms may lead to a permanent ban.
+
+### 4. Permanent Ban
+
+Community Impact: Demonstrating a pattern of violation of community standards, including sustained inappropriate behavior, harassment of an individual, or aggression toward or disparagement of classes of individuals.
+
+Consequence: A permanent ban from any sort of public interaction within the community.
```
Comment (edmofro) [inline .github/code-of-conduct.md:71]: I think we can simplify these, given how immature and small our community is likely to be. Let's just go for "1. Warning" and "2. Temporary or Permanent Ban". The first can have the content from the existing "Correction", and the second the content from the "Temporary Ban" with a tweak:

"A ban from any sort of interaction or public communication with the community, either permanently or for a specified period of time. No public or private interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct, is allowed during this period."

**Review by edmofro — CHANGES_REQUESTED** [review]
Thanks @acdunham!

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (acdunham) [reply .github/CONTRIBUTING.md:19]: I actually copied this from Michelle's and it was reviewed by Felix, so I assume that's right? But also happy to remove

**Review by edmofro — APPROVED** [review]
lgtm! 
I think leave that CLA thing as is, and we'll try to make it work (following [here](https://beyondessential.slack.com/archives/C06AJQE49B8/p1703198842420789?thread_ts=1702958432.659259&cid=C06AJQE49B8) )

**Pull request merged**